### PR TITLE
Disabled suppressions of 1801, 1804, and 1823

### LINKED ISSUE #1103: CA1801 should consider parameter being used in if statement
I have this error

Warning	CA1801	Parameter symbol of method IsImplementationOfAnyExplicitInterfaceMember is never used. Remove the parameter or use it in the method body.	

for this code

```
public static bool IsImplementationOfAnyExplicitInterfaceMember(this ISymbol symbol)
        {
            if (symbol is IMethodSymbol methodSymbol && methodSymbol.ExplicitInterfaceImplementations.Any())
            {
                return true;
            }

            if (symbol is IPropertySymbol propertySymbol && propertySymbol.ExplicitInterfaceImplementations.Any())
            {
                return true;
            }

            if (symbol is IEventSymbol eventSymbol && eventSymbol.ExplicitInterfaceImplementations.Any())
            {
                return true;
            }

            return false;
        }
```

### PR DESCRIPTION
**Pull request by 333fred**
The underlying IOp issues have been fixed. Fixes #1103, #933, and #935. Tagging @dotnet/roslyn-analysis for review.

### CODE REVIEW

**Comment by jinujoseph** [comment]
993 ? wrong id ? you mean 933 ?

File: src/Microsoft.CodeAnalysis.FxCopAnalyzers/Setup/FxCopAnalyzersPackage.cs [hunk 1]
```diff
@@ -11,8 +11,6 @@ namespace FxCopAnalyzersSetup
     class FxCopAnalyzersPackage : Package
 #pragma warning restore CA1812
     {
-#pragma warning disable CA1823 // False unused field diagnostic: https://github.com/dotnet/roslyn-analyzers/issues/1191
```
Comment (jinujoseph) [inline src/Microsoft.CodeAnalysis.FxCopAnalyzers/Setup/FxCopAnalyzersPackage.cs]: good to close https://github.com/dotnet/roslyn-analyzers/issues/1191 ?

Comment (333fred) [inline src/Microsoft.CodeAnalysis.FxCopAnalyzers/Setup/FxCopAnalyzersPackage.cs]: I believe so, looks like a dupe of https://github.com/dotnet/roslyn-analyzers/issues/933.

**Comment by 333fred** [comment]
@dotnet/roslyn-infrastructure @mmitche looks like a java exception when attempting to archive files: https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/windows_release_prtest/201/console.

**Comment by 333fred** [comment]
@dotnet-bot retest prtest/windows/release please

**Comment by 333fred** [comment]
Now additionally fixes https://github.com/dotnet/roslyn-analyzers/issues/1204.

**Comment by 333fred** [comment]
@dotnet-bot retest this please

**Comment by 333fred** [comment]
@dotnet/roslyn-infrastructure unauthorized to delete issues.
https://ci.dot.net/job/dotnet_roslyn-analyzers/job/master/job/windows_debug_prtest/206/console,
```
14:29:28 Remove-Item : Cannot remove item C:\Users\dotnet-bot\.nuget\packages\microsoft.net.compilers\2.3.0-beta2-61719-05\tools
14:29:28 \Microsoft.Build.Tasks.CodeAnalysis.dll: Access to the path 'Microsoft.Build.Tasks.CodeAnalysis.dll' is denied.
14:29:28 At D:\j\workspace\windows_debug---c729d7ab\build\Build.ps1:83 char:3
14:29:28 +   Remove-Item $nugetRoot -Recurse -Force
14:29:28 +   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
14:29:28     + CategoryInfo          : PermissionDenied: (Microsoft.Build.Tasks.CodeAnalysis.dll:FileInfo) [Remove-Item], Unaut 
14:29:28    horizedAccessException
14:29:28     + FullyQualifiedErrorId : RemoveFileSystemItemUnAuthorizedAccess,Microsoft.PowerShell.Commands.RemoveItemCommand
```

**Pull request merged**